pkg/relayer/interface.go

```diff
@@ -27,10 +27,21 @@ type RelayerProxy interface {
 
 	// VerifyRelayRequest is a shared method used by RelayServers to check the
 	// relay request signature and session validity.
-	VerifyRelayRequest(relayRequest *types.RelayRequest) (isValid bool, err error)
-
-	// SignRelayResponse is a shared method used by RelayServers to sign the relay response.
-	SignRelayResponse(relayResponse *types.RelayResponse) ([]byte, error)
+	// This method may mutate the relay request in the future, so the returned
```

Comment:
I think it's kind of confusing that:
1. We pass in a pointer to an object
2. That pointer may be mutated
3. We return it

I believe it should be one of:
1. Pass a pointer that may be mutated
2. Pass a value that's copied, mutated, and returned

Reply:
I see... I'll revert to passing the pointer without returning and mutate it inside the function as I can't clearly foresee the impact of changing the whole workflow to not use pointers. We'll keep that question to when we start implementing the middleware capabilities.

Reply:
cc @bryanchriswhite